model/riscv_sys_regs.sail

```diff
@@ -673,39 +671,34 @@ function clause read_CSR(0xF15) = mconfigptr
 /* S-mode registers */
 
 /* sstatus reveals a subset of mstatus */
-bitfield Sstatus : xlenbits = {
-  SD   : xlen - 1,
-  // The UXL field does not exist on RV32, so we define an explicit
-  // getter and setter below.
-  // UXL  : 30 .. 29,
-  MXR  : 19,
-  SUM  : 18,
-  XS   : 16 .. 15,
-  FS   : 14 .. 13,
-  VS   : 10 .. 9,
-  SPP  : 8,
-  SPIE : 5,
-  SIE  : 1,
+bitfield Sstatus : bits(64) = {
+  SD_64 : 63,
+
+  UXL   : 33 .. 32,
+  SD_32 : 31,
+  SDT   : 24,
+  SPELP : 23,
+  MXR   : 19,
+  SUM   : 18,
+  XS    : 16 .. 15,
+  FS    : 14 .. 13,
+  VS    : 10 .. 9,
+  SPP   : 8,
+  SPIE  : 5,
+  SIE   : 1,
 }
 /* sstatus is a view of mstatus, so there is no register defined. */
 
-function get_sstatus_UXL(s : Sstatus) -> arch_xlen = {
-  let m = Mk_Mstatus(s.bits);
-  get_mstatus_UXL(m)
-}
-
-function set_sstatus_UXL(s : Sstatus, a : arch_xlen) -> Sstatus = {
-  let m = Mk_Mstatus(s.bits);
-  let m = set_mstatus_UXL(m, a);
-  Mk_Sstatus(m.bits)
-}
-
 function lower_mstatus(m : Mstatus) -> Sstatus = {
   let s = Mk_Sstatus(zeros());
-  let s = set_sstatus_UXL(s, get_mstatus_UXL(m));
+  let sd = get_mstatus_SD(m);
 
   [s with
-    SD = m[SD],
+    SD_64 = if xlen == 64 then sd else 0b0,
```

Comment:
I think I would just do `SD_64 = m[SD_64]` and `SD_32 = m[SD_32]`. The rest of the code should ensure that they don't get set incorrectly.